src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/v4/ServiceResourceApiImpl.java

```diff
@@ -57,6 +62,13 @@ public ResponseEntity<Service> createService(ServiceCreate service) {
             service.getServiceCharacteristic().forEach(ch -> params.put(ch.getName(), ch.getValue()));
         }
         ofNullable(service.getState()).ifPresent(state -> params.put(STATE, state));
+        if (isNotEmpty(service.getServiceOrderItem())) {
```

Comment:
Please extract params collection to separate private method. There are 3 types of collection
`if (isNotEmpty(service.getRelatedParty()))`
`isNotEmpty(service.getServiceCharacteristic())`
`(isNotEmpty(service.getServiceOrderItem()))`

method 
`HashMap collectContext(ServiceCreate service)`
could collect and return saga context.

Reply:
Fixed